### ISSUE TITLE
WireGuard configuration

### ISSUE THREAD

**Issue by gsanchietti** [NethServer/nethsecurity#921]
**Description:**
This feature introduces WireGuard configuration from both the command line interface (CLI) and the user interface (UI) for the NethSecurity project. It allows administrators to manage multiple WireGuard server instances and their associated peers (accounts) with specific configurations and features.

#### Features:
- Support for multiple WireGuard server instances
- Each instance has its own reserved zone
- Static IP allocation for all peers (only /24 networks are supported)
- Ability to download peer configuration as text or QR code
- Net-to-net (site-to-site) mode
- Support for peer pre-shared keys
- Link a local or remote user to a given WireGuard account
- Use WireGuard account IP addresses within firewall objects
- Add an API to import configuration files for firewalls used as peers in a net-to-net setup
- Support for UI to create multiple instances of servers
  - If a server is associated with a user database, all accounts must be tied to existing users (similar to OpenVPN RW)
  - If a server is not associated with a user database, it is possible to create custom WireGuard-only users, typically used for connecting multiple sites in point-to-point or star topology
- Support kernel-level debug

#### References:
- [Pull Request #888](https://github.com/NethServer/nethsecurity/pull/888)
- [WireGuard iOS App](https://apps.apple.com/it/app/wireguard/id1441195209)
- [WireGuard Android App](https://play.google.com/store/apps/details?id=com.wireguard.android&hl=it&pli=1)

**Comment by cotosso** [comment]
TEST CASE  for **"Create the instance"**

```
root@NSec8-VM:~# echo '{"listen_port": 51820, "name": "wg1", "instance": "wg1", "enabled": true, "network": "10.103.1.0/24", "routes": ["192.168.100.0/24"], "public_endpoint": "1.2.3.4", "dns": [], "user_db": ""}' |  /usr/libexec/rpcd/ns.wireguard call set-instance
{"public_key": "10VNMyuL/4kZz2IGnxlCBFggtxgrXP1IifVZGfuoaBE="}
```

instance has been created but we need some command more to see the wg interface

```
uci commit network && uci commit firewall
reload_config
```


and then everything is fine

```
10: wg1: <POINTOPOINT,NOARP,UP,LOWER_UP> mtu 1420 qdisc noqueue state UNKNOWN group default qlen 1000
    link/none
    inet 10.103.1.1/32 brd 255.255.255.255 scope global wg1
       valid_lft forever preferred_lft forever
```

**Comment by cotosso** [comment]
Test Case for **"Add a new account (peer)"**
```
root@NSec8-VM:~# echo '{"enabled": true, "instance": "wg1", "account": "user1", "route_all_traffic": false, "client_to_client": false, "ns_routes": [], "preshared_key": true}' | /usr/libexec/rpcd/ns.wireguard call set-peer
{"section": "wg1_user1_peer"}
```

some commit and restart:
```
root@NSec8-VM:~# uci commit network
root@NSec8-VM:~# reload_config
```

and then I can see the new account inside the configuration

```
config wireguard_wg1 'wg1_user1_peer'
        option public_key '3Kirpj+r70ZEng0ZLCp3/X8/V7rFHRGB+66SQ8SgDDw='
        option private_key 'WOepFqyNkpg3C5i+KxVxwMnBhkJn2m/W3/BCwrvg4Xg='
        option ns_ip '10.103.1.2'
        option persistent_keepalive '25'
        option description 'user1'
        option ns_link 'network/wg1'
        option route_allowed_ips '1'
        option disabled '0'
        option ns_route_all_traffic '0'
        option ns_client_to_client '0'
        list allowed_ips '10.103.1.2'
        option preshared_key 'KiXANB3TIisUgdlLvrC0EkxDCMy7SRiI2UqCk8oNmAk='
```

**Comment by gsanchietti** [comment]
> Environment: Virtual machine going to internet through another firewall.

That's the issue: probably the firewall has DNS proxy configured that intercepts the request.

I'm setting it as verified.

**Comment by cotosso** [comment]
TEST CASE for **"Download the account configuration"**

Everything went fine

```
root@NSec8-VM:~# echo '{"instance": "wg1", "account": "user1"}' | /usr/libexec/rpcd/ns.wireguard call download-peer-config | jq -r .config
# Account: user1 for wg1
[Interface]
PrivateKey = WOepFqyNkpg3C5i+KxVxwMnBhkJn2m/W3/BCwrvg4Xg=
Address = 10.103.1.2
# Custom DNS disabled

[Peer]
PublicKey = 10VNMyuL/4kZz2IGnxlCBFggtxgrXP1IifVZGfuoaBE=
PreSharedKey = KiXANB3TIisUgdlLvrC0EkxDCMy7SRiI2UqCk8oNmAk=
AllowedIPs = 192.168.100.0/24,10.103.1.1
Endpoint = 1.2.3.4:51820
PersistentKeepalive = 25
```

**Comment by cotosso** [comment]
TEST CASE  for **"Configure a Desktop/Mobile device"**

Everything went fine:
QR code has been generated and correctly scanned/importaed by the wireguard android app.

**Comment by gsanchietti** [comment]
Released on 23.05.5-ns.1.4.0
